### PR TITLE
Add options to erizo build process

### PR DESCRIPTION
**Pull request by lodoyun**
**Description**

This PR adds "Release" and "Debug" configurations for both erizo and erizoAPI. The main difference is that Erizo is now built with `-O3`. Both are built by default, while "Release" is the one that's normally used by ErizoJS.
In order to run the debug version of both erizo and erizoAPI, you can start erizoAgent with `initErizoAgent.sh -d` or, if you are running the docker image, you can add `--erizoDebug`.

Also, we're now using `rpath` when building erizoAPI, meaning that we no longer need `LD_LIBRARY_PATH` for Licode.

**Warning for Spine users**: You do need `LD_LIBRARY_PATH` and note that `liberizo` is now under `erizo/build/release/erizo` or `erizo/build/debug/erizo` .

Finally, in the Docker image we're keeping both builds but removing the object files in `CMakeFiles`to save some space.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: extras/docker/initDockerLicode.sh [hunk 1]
```diff
@@ -115,7 +119,11 @@ run_erizoController() {
 run_erizoAgent() {
   echo "Starting erizoAgent"
   cd $ROOT/erizo_controller/erizoAgent
-  node erizoAgent.js &
+  if [ "$ERIZODEBUG" = "true" ]; then
```
Comment (jcague) [inline extras/docker/initDockerLicode.sh]: `==` or `-eq` ?

File: erizoAPI/binding.gyp [hunk 2]
```diff
@@ -1,10 +1,38 @@
...
+              'GCC_ENABLE_CPP_RTTI': 'YES',              # -fno-rtti
+              'MACOSX_DEPLOYMENT_TARGET' : '10.11',      #from MAC OS 10.7
+              'OTHER_CFLAGS': [
+              '-g -O3 -stdlib=libc++ -std=c++11',
+            ]
+          },
+        }, { # OS!="mac"
+          'cflags!' : ['-fno-exceptions'],
+          'cflags' : ['-D__STDC_CONSTANT_MACROS'],
+          'cflags_cc' : ['-Wall', '-O3', '-g' , '-std=c++11', '-fexceptions'],
```
Comment (jcague) [inline erizoAPI/binding.gyp]: :+1:

File: erizo/src/erizo/media/ExternalInput.cpp [hunk 3]
```diff
@@ -59,7 +59,8 @@ int ExternalInput::init() {
 
   // VideoCodecInfo info;
   MediaInfo om;
-  AVStream *st, *audio_st;
+  AVStream *st = nullptr;
```
Comment (jcague) [inline erizo/src/erizo/media/ExternalInput.cpp]: good catch!

File: scripts/installErizo.sh [hunk 4]
```diff
@@ -53,6 +55,9 @@ install_erizo(){
   cd $ROOT/erizo
   ./generateProject.sh
   ./buildProject.sh $FAST_MAKE
+  if [ $DELETE_OBJECT_FILES == 'true' ]; then
```
Comment (jcague) [inline scripts/installErizo.sh]: in other places we do `if [ "$DELETE_OBJECT_FILES" == "true" ];`, I didn't know it works like this too, great!

File: extras/docker/initDockerLicode.sh (hunk 1 above)
Comment (lodoyun) [reply extras/docker/initDockerLicode.sh]: We're using `=` throughout the whole file but `==` is more POSIX compliant so I'll switch all of them 👍

File: scripts/installErizo.sh (hunk 4 above)
Comment (lodoyun) [reply scripts/installErizo.sh]: Even if it works, I'll switch to `"` for consistency 🙂

**Pull request merged**